### PR TITLE
Avoid Double WIA

### PR DESCRIPTION
**Pull request by adghayes**
## Overview

VVSG 2.0 1.1.4-I:

> The voting system must be capable of gathering and recording write-in votes within a voting process that allows for reconciliation of aliases and double votes.

VVSG requires us to account for the possibility of double votes via write-ins. In this pull request, I make it impossible to create adjudications that would create a double vote. I chose to do so via a pop-up method rather than simply disabling or re-arranging buttons. I lean toward a) the buttons staying in the same places for every ballot and b) providing more information than simply being disabled.

## Demo Video or Screenshot

https://github.com/votingworks/vxsuite/assets/37960853/a5efb986-0750-4881-aa4d-f877e0d29b13

## Testing Plan
- updated and new automated tests
- manual testing

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: apps/admin/backend/src/util/write_ins.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+  return {
+    imageUrl: toDataUrl(await loadImageData(image), 'image/jpeg'),
+    ballotCoordinates: {
+      ...layout.pageSize,
+      x: 0,
+      y: 0,
+    },
+    contestCoordinates: contestLayout.bounds,
+    writeInCoordinates: writeInLayout.bounds,
```
Comment (adghayes) [inline apps/admin/backend/src/util/write_ins.ts:100]: Copied over from `app.ts`.

File: apps/admin/frontend/src/screens/write_ins_adjudication_screen.tsx [hunk 2]
```diff
@@ -332,6 +403,88 @@ export function WriteInsAdjudicationScreen({
       nextButton.current?.focus();
     }, 0);
   }
+
+  const isWriteInDetailViewFresh =
+    writeInDetailViewQuery.isSuccess && !writeInDetailViewQuery.isStale;
```
[the remark this answers is not in the document]
Comment (adghayes) [reply apps/admin/frontend/src/screens/write_ins_adjudication_screen.tsx:408]: I plan to put the double vote check on the backend but going to hold off for now, as I'm going to play with the data model in subsequent PRs that would force me to rewrite it anyway.

**Comment by kofi-q** [comment]
One UX flow nit that didn't occur to me when we sync'ed previously: when the modal pops up, is there another action we can offer to the user?

IIUC, the only possible actions left are to mark the write-in as invalid, or choose a different candidate (in which case, the user probably made a mistake clicking the already-bubbled/already-adjudicated vote) - if that's the case, could the modal have a "Mark write-in invalid" primary action?

Or at least, the modal could provide guidance that if it's truly a duplicate vote, they should mark the write-in as invalid?

**Comment by adghayes** [comment]
Although it makes sense to have more guidance in the text, I was hesitant to provide any guidance. The only option they have in our system if it truly is a double vote is to mark it as invalid, so that would make be the primary action if there were one. But a) everything in WIA will be so jurisdiction specific that I don't want to be descriptive if I don't have to be and b) if they arrived at the modal by clicking the wrong candidate, which seems more likely to me than a true double vote, then I'm not sure I want to give them a primary action at all. I think if a write-in is truly a double-vote, they're going to want to take a closer look before marking as invalid.

@mattroe do you have any additional thoughts on the UX?

File: apps/admin/frontend/src/screens/write_ins_adjudication_screen.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kofi-q) [reply apps/admin/frontend/src/screens/write_ins_adjudication_screen.tsx:408]: +1 to disabling the buttons during server requests

**Review by kofi-q — APPROVED** [review]
Code and UX LGTM - just that one remaining open question on providing more user guidance for double votes, but that doesn't blocking either.

**Comment by adghayes** [comment]
modals with updated copy:

Screen Shot 2023-05-10 at 12 10 50 PM
Screen Shot 2023-05-10 at 12 11 00 PM

**Pull request merged**